src/app/views/query-runner/request/permissions/PermissionItemV9.tsx

```diff
@@ -0,0 +1,168 @@
+import {
+  Button,
+  Tooltip,
+  Label,
+  makeStyles,
+  tokens
+} from '@fluentui/react-components';
+import { useAppDispatch, useAppSelector } from '../../../../../store';
+import { IPermission, IPermissionGrant } from '../../../../../types/permissions';
+import { revokeScopes } from '../../../../services/actions/revoke-scopes.action';
+import { REVOKING_PERMISSIONS_REQUIRED_SCOPES } from '../../../../services/graph-constants';
+import { consentToScopes } from '../../../../services/slices/auth.slice';
+import { getAllPrincipalGrant, getSinglePrincipalGrant } from '../../../../services/slices/permission-grants.slice';
+import { translateMessage } from '../../../../utils/translate-messages';
+import { PermissionConsentType } from './ConsentType';
+import { InfoRegular } from '@fluentui/react-icons';
+
+interface PermissionItemProps {
+  item: IPermission;
+  index: number;
+  column: { key: string; fieldName?: string } | undefined;
+}
+
+const useStyles = makeStyles({
+  adminLabel: {
+    fontSize: tokens.fontSizeBase300,
+    padding: tokens.spacingVerticalXS
+  },
+  button: {
+    margin: tokens.spacingVerticalXXS
+  },
+  tooltip: {
+    display: 'block'
+  },
+  icon: {
+    position: 'relative',
+    top: '4px',
+    cursor: 'pointer'
+  }
+});
+
+const PermissionItem = (props: PermissionItemProps): JSX.Element | null => {
+  const dispatch = useAppDispatch();
+  const { scopes, auth: { consentedScopes }, profile: { user }, permissionGrants } = useAppSelector((state) => state);
```

Comment:
Get the specific state values to avoid rerenders in the component.

```suggestion
const scope = useAppSelector(state => state.scope)
// the other values
```